src/Unity.Mathematics/Extras/NormalizedPlane.cs

```diff
@@ -0,0 +1,124 @@
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace Unity.Mathematics.Extras
+{
+    /// <summary>
+    /// A plane represented by a unit length normal vector and a distance along the normal from the origin.
+    /// </summary>
+    /// <remarks>
+    /// A plane splits the 3D space in half.  The normal vector points to the positive half and the other half is
+    /// considered negative.
+    /// </remarks>
+    // A plane described by a normal and a distance from the origin
+    [DebuggerDisplay("{Normal}, {Distance}")]
+    [Serializable]
+    internal struct NormalizedPlane
+    {
+        /// <summary>
+        /// A plane in the form Ax + By + Cz + Dw = 0.
+        /// </summary>
+        /// <remarks>
+        /// Stores the coefficients A, B, C, D where (A, B, C) is a unit length normal and D is distance from the
+        /// origin along the normal.
+        /// </remarks>
+        public float4 NormalAndDistance;
+
+        /// <summary>
+        /// Constructs a plane with a normal vector and distance from the origin.
+        /// </summary>
+        /// <param name="normal">A non-zero vector that is perpendicular to the plane.  It may be any length.</param>
+        /// <param name="distance">Distance from the origin along the normal.  A negative value moves the plane in the
+        /// same direction as the normal while a positive value moves it in the opposite direction.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public NormalizedPlane(float3 normal, float distance)
+        {
+            NormalAndDistance = new float4(math.normalize(normal), distance);
+        }
+
+        /// <summary>
+        /// Constructs a plane with a normal vector and a point that lies in the plane.
+        /// </summary>
+        /// <param name="normal">A non-zero vector that is perpendicular to the plane.  It may be any length.</param>
+        /// <param name="pointInPlane">A point that lies in the plane.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public NormalizedPlane(float3 normal, float3 pointInPlane)
+        {
+            NormalAndDistance = new float4(normal, -math.dot(normal, pointInPlane));
+        }
+
+        /// <summary>
+        /// Constructs a plane with two vectors and a point that all lie in the plane.
+        /// </summary>
+        /// <param name="vector1InPlane">A non-zero vector that lies in the plane.  It may be any length.</param>
+        /// <param name="vector2InPlane">A non-zero vector that lies in the plane.  It may be any length and must be different from <paramref name="vector1InPlane"/>.</param>
```

Comment:
```suggestion
        /// <param name="vector2InPlane">A non-zero vector that lies in the plane.  It may be any length and must not be a scalar multiple of <paramref name="vector1InPlane"/>.</param>
```
Not sure exactly how to word this or how much linear algebra you want to discuss in the docstring. But passing (1, 0, 0) and (-1, 0, 0) are "different", but they won't give you a very good plane.